src/artemis/experiment_plans/tests/conftest.py

```diff
@@ -178,15 +194,45 @@ def fake_create_devices(
     smargon.omega.set = mock_omega_sets
 
     devices = {
-        "eiger": i03.eiger(fake_with_ophyd_sim=True),
+        "eiger": eiger,
         "smargon": smargon,
         "zebra": zebra,
-        "detector_motion": i03.detector_motion(fake_with_ophyd_sim=True),
+        "detector_motion": detector_motion,
         "backlight": i03.backlight(fake_with_ophyd_sim=True),
     }
     return devices
 
 
+@pytest.fixture()
+def fake_create_rotation_devices(
+    eiger: EigerDetector,
+    smargon: Smargon,
+    zebra: Zebra,
+    detector_motion: DetectorMotion,
+    backlight: Backlight,
+    attenuator: Attenuator,
+):
+    eiger.stage = MagicMock()
+    eiger.unstage = MagicMock()
+    mock_omega_sets = MagicMock(return_value=Status(done=True, success=True))
+
+    mock_arm_disarm = MagicMock(
+        side_effect=zebra.pc.arm.armed.set, return_value=Status(done=True, success=True)
+    )
+    zebra.pc.arm.set = mock_arm_disarm
+    smargon.omega.velocity.set = mock_omega_sets
+    smargon.omega.set = mock_omega_sets
```

Comment:
Should: This feels wrong, it makes the tests much more complicated as it's now one mock for both position and velocity. I think separate the mocks then you can assert things more nicely